src/server/mod.rs

```diff
@@ -25,6 +25,7 @@ use crate::types::RedisPool;
 pub struct SpaceapiServerBuilder {
     status: api::Status,
     redis_connection_info: Result<ConnectionInfo, SpaceapiServerError>,
+    pool: Option<r2d2::Pool<r2d2_redis::RedisConnectionManager>>,
```

Comment:
Maybe we should just have the `pool` in the builder and just allow creating it by either providing it directly or providing a `ConnectionInfo`. So let's just store a `Result<r2d2::Pool<r2d2_redis::RedisConnectionManager>, SpaceapiServerError>` so we are able to return any error on `build`.

Reply:
Hm, my rationale was to avoid creating the pool in `.redis_connection_info`, but this way is probably cleaner...